src/jit/importer.cpp

```diff
@@ -17565,6 +17565,14 @@ void Compiler::impMarkInlineCandidate(GenTreePtr             callNode,
     GenTreeCall* call = callNode->AsCall();
     InlineResult inlineResult(this, call, nullptr, "impMarkInlineCandidate");
 
+#ifdef _TARGET_ARM_
+    // When inlining methods that return type is struct, 
+    // legacy JIT for ARM fails to assign correct value for struct return
+    if (call->gtReturnType == TYP_STRUCT) {
```

Comment:
Isn't it problematic only for HFA structs? If so, we need to something like `IsHfa(call->gtRetClsHnd)` in the if condition.